announcements.py

```diff
@@ -1,61 +1,92 @@
 import datetime
+import json
 import time
-# import requests
+import requests
+import itertools
 from multiprocessing import Process
 
-class MinsBeforeAnnouncement:
-    def __init__(self, minsBefore, text):
-        self.minsBefore = minsBefore
+class Announcement:
+    id_iter = itertools.count()
+    def __init__(self, time, text, banana_time):
+        self.id = next(self.id_iter)
+        self.time = time
         self.text = text
+        self.banana_time = banana_time
 
-class MinsBeforeAnnouncementWorker(Process):
-    def __init__(self, minsBefore, text, time):
-        super(MinsBeforeAnnouncementWorker, self).__init__()
-        self.minsBefore = minsBefore
-        self.text = text
+class AnnouncementWorker(Process):
+    def __init__(self, id, time, text, banana_time):
+        super(AnnouncementWorker, self).__init__()
+        self.id = id
         self.time = time
+        self.text = text
+        self.banana_time = banana_time
 
     def run(self):
         while True:
-            currentTime = datetime.datetime.now()
-            normalisedTime = datetime.datetime.combine(datetime.date.today(), self.time)
-            alertTime = normalisedTime - datetime.timedelta(minutes = self.minsBefore)
+            current_time = datetime.datetime.now()
             
-            sleepTime = alertTime - currentTime
-            sleepTimeSeconds = sleepTime.seconds
-            print("[" + str(datetime.datetime.now()) + "] DEBUG - Announcement for " + str(self.minsBefore) + " minutes before is sleeping for " + str(sleepTimeSeconds) + " seconds")
-            time.sleep(sleepTimeSeconds)
+            alert_time = datetime.datetime.combine(datetime.date.today(), self.time)
+            
+            sleep_time = alert_time - current_time
+            sleep_time_seconds = sleep_time.seconds
+            print("[" + str(datetime.datetime.now()) + "] DEBUG - Announcement for " + str(self.time) + " sleeping for seconds: " + str(sleep_time_seconds) + " seconds")
```

Comment:
Make these `print`s use the logger if possible

Reply:
Multiprocessing logging looks like it is a bit more complex than standard logging. There is some information about a library here: https://stackoverflow.com/questions/641420/how-should-i-log-while-using-multiprocessing-in-python